internal/bminventory/inventory.go

```diff
@@ -2263,7 +2263,9 @@ func (b *bareMetalInventory) updateDhcpNetworkParams(updates map[string]interfac
 		log.WithError(err).Warnf("Set Ingress VIP")
 		return common.NewApiError(http.StatusBadRequest, err)
 	}
-	if params.ClusterUpdateParams.MachineNetworks != nil {
+	// VIPs are always allocated from the first provided machine network. We want to trigger
+	// their reallocation only when this network has changed, but not in other cases.
+	if params.ClusterUpdateParams.MachineNetworks != nil && string(params.ClusterUpdateParams.MachineNetworks[0].Cidr) != primaryMachineCIDR {
```

Comment:
I'm not 100% convinced about accessing `params.ClusterUpdateParams.MachineNetworks[0].Cidr` without explicitly checking its existence... In https://github.com/openshift/assisted-service/pull/2512 I'm adding more validations so that it should be much more difficult for the object like this to slip through, but still, opinions appreciated.

Reply:
You have to check for `nil` first, this is just fine.